cob_twist_controller/src/damping_methods/damping.cpp

```diff
@@ -118,20 +127,50 @@ double DampingManipulability::getDampingFactor(const Eigen::VectorXd& sorted_sin
 /**
  * Method returns the damping factor according to the least singular value.
  */
-double DampingLeastSingularValues::getDampingFactor(const Eigen::VectorXd& sorted_singular_values,
+Eigen::MatrixXd DampingLeastSingularValues::getDampingFactor(const Eigen::VectorXd& sorted_singular_values,
                                                     const Eigen::MatrixXd& jacobian_data) const
 {
     // Formula 15 Singularity-robust Task-priority Redundandancy Resolution
     double least_singular_value = sorted_singular_values(sorted_singular_values.rows() - 1);
+    uint32_t rows = sorted_singular_values.rows();
+    Eigen::MatrixXd lambda = Eigen::MatrixXd::Identity(rows,rows);
     if (least_singular_value < this->params_.eps_damping)
     {
         double lambda_quad = pow(this->params_.lambda_max, 2.0);
-        return sqrt( (1.0 - pow(least_singular_value / this->params_.eps_damping, 2.0)) * lambda_quad);
+        //lambda(sorted_singular_values.rows() - 1,sorted_singular_values.rows() - 1)= sqrt( (1.0 - pow(least_singular_value / this->params_.eps_damping, 2.0)) * lambda_quad);
```

Comment:
Please remove this line as it is a remnant from when the "damping factor only for last index" times....